apps/literatureclock/literature_clock.star

```diff
@@ -0,0 +1,180 @@
+"""
+Applet: Literature Clock
+Summary: Quotes containing the time
+Description: Displays the time using a quote from a piece of literature. Based on work and idea by Jaap Meijers and Jene Voldsen.
+Author: Alysha Kwok
+"""
+
+load("encoding/json.star", "json")
+load("http.star", "http")
+load("random.star", "random")
+load("render.star", "render")
+load("schema.star", "schema")
+load("time.star", "time")
+
+DEFAULT_LOCATION = """
+{
+	"lat": "40.6781784",
+	"lng": "-73.9441579",
+	"description": "Brooklyn, NY, USA",
+	"locality": "Brooklyn",
+	"place_id": "ChIJCSF8lBZEwokRhngABHRcdoI",
+	"timezone": "America/New_York"
+}
+"""
+JSON_ENDPOINT = "https://raw.githubusercontent.com/alyshakwok/literature-clock/master/docs/times/"
+QUOTE_FIRST = "quote_first"
+QUOTE_TIME = "quote_time_case"
+QUOTE_LAST = "quote_last"
+TITLE = "title"
+AUTHOR = "author"
+SFW = "sfw"
+DEFAULT_FONT = "tb-8"
+SMALL_FONT = "tom-thumb"
+
+def get_data(fileTime, config):
+    file = JSON_ENDPOINT
+    if fileTime.hour < 10:
+        file += "0"
+    file += str(fileTime.hour) + "_"
+    if fileTime.minute < 10:
+        file += "0"
+    file += str(fileTime.minute) + ".json"
+
+    # check if quotes is not empty, else one min back
+    request = http.get(file)
```

Comment:
It'd be great if you could add some caching around this call, to keep request rates down. Check out the `cache.star` module.

Reply:
Added in some caching!